src/backend/api/Fusion.Resources.Logic/Requests/CanApproveStepHandlerBase.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Fusion.AspNetCore.FluentAuthorization;
+using Fusion.Authorization;
+using Fusion.Resources.Api.Authorization.Requirements;
+using Fusion.Resources.Authorization.Requirements;
+using Fusion.Resources.Database.Entities;
+using Fusion.Resources.Domain;
+using Microsoft.AspNetCore.Authorization;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Logic.Requests
+{
+    public record WorkflowAccessKey(string Subtype, string CurrentStep)
+    {
+        public static implicit operator WorkflowAccessKey((string, string) tuple)
+            => new WorkflowAccessKey(tuple.Item1, tuple.Item2);
+    }
+    public record WorkflowAccess()
+    {
+        public bool IsResourceOwnerAllowed { get; init; } = false;
+        public bool IsParentResourceOwnerAllowed { get; init; } = false;
+        public bool IsSiblingResourceOwnerAllowed { get; init; } = false;
+        public bool IsAllResourceOwnersAllowed { get; init; } = false;
+        public bool IsCreatorAllowed { get; init; } = false;
+        public bool IsDirectTaskOwnerAllowed { get; init; } = false;
+        public bool IsOrgChartTaskOwnerAllowed { get; init; } = false;
+        public bool IsOtherProjectMembersAllowed { get; init; } = false;
+        public bool IsOrgChartReadAllowed { get; init; } = false;
+        public bool IsOrgChartWriteAllowed { get; init; } = false;
+        public bool IsOrgAdminAllowed { get; init; } = false;
+
+        public static WorkflowAccess Default = new WorkflowAccess();
+    }
+
+    public class CanApproveStepHandlerBase
+    {
+        private readonly IAuthorizationService authorizationService;
+
+        public CanApproveStepHandlerBase(IAuthorizationService authorizationService)
+        {
+            this.authorizationService = authorizationService;
+        }
+        protected async Task CheckAccess(DbResourceAllocationRequest request, WorkflowAccess row, ClaimsPrincipal initiator)
+        {
+            bool isAllowed = false;
+
+            if (!string.IsNullOrEmpty(request.AssignedDepartment))
```

Comment:
Could we use the fluent pattern here so we use the `if`'s to build the rule, then only evaluate once.. That way we only evaluate to the first success..